cedar-policy-core/src/ast/request.rs

```diff
@@ -186,6 +197,19 @@ impl Request {
     pub fn context(&self) -> Option<&Context> {
         self.context.as_ref()
     }
+
+    /// Get the request types that correspond to this request.
+    /// This includes the types of the principal, action, and resource.
+    /// [`RequestType`] is used by the entity manifest.
+    /// The context type is implied by the action's type.
+    /// Returns `None` if the request is not fully concrete.
+    pub fn to_request_type(&self) -> Option<RequestType> {
+        Some(RequestType {
+            principal: self.principal.uid()?.clone().components().0,
```

Comment:
nit: `.clone().components().0` will result in cloning both components, whereas `.entity_type().clone()` only clones the component it needs